editor-components/slider/slider10/slider10.tsx

```diff
@@ -0,0 +1,470 @@
+        },
+        {
+          type: "page",
+          key: "link",
+          displayer: "Link",
+          value: ""
+        }
+      ]
+    });
+    this.addProp({
+      type: "object",
+      key: "second-featured-item",
+      displayer: "Second Featured Item",
+      value: [
+        {
+          type: "string",
+          key: "title",
+          displayer: "Title",
+          value: "The Arches"
+        },
+        {
+          type: "string",
+          key: "subtitle",
+          displayer: "Subtitle",
+          value: "Modern House"
+        },
+        {
+          type: "image",
+          key: "image",
+          displayer: "Image",
+          value: "https://storage.googleapis.com/download/storage/v1/b/hq-composer-0b0f0/o/668bb90fba6bbe002b64588b?alt=media"
+        },
+        {
+          type: "page",
+          key: "link",
+          displayer: "Link",
+          value: ""
+        }
+      ]
+    });
+    this.addProp({
+      type: "object",
+      key: "third-featured-item",
+      displayer: "Third Featured Item",
+      value: [
+        {
+          type: "string",
+          key: "title",
+          displayer: "Title",
+          value: "Dream House"
+        },
+        {
+          type: "string",
+          key: "subtitle",
+          displayer: "Subtitle",
+          value: "Modern Farmhouse"
+        },
+        {
+          type: "image",
+          key: "image",
+          displayer: "Image",
+          value: "https://storage.googleapis.com/download/storage/v1/b/hq-composer-0b0f0/o/668bb921ba6bbe002b645896?alt=media"
+        },
+        {
+          type: "page",
+          key: "link",
+          displayer: "Link",
+          value: ""
+        }
+      ]
+    });
+
+    this.addProp({
+      type: "icon",
+      key: "prev-button-icon",
+      displayer: "Previous Slide Button",
+      value: "FaArrowLeft"
+    });
+    this.addProp({
+      type: "icon",
+      key: "next-button-icon",
+      displayer: "Next Slide Button",
+      value: "FaArrowRight"
+    });
+  }
+
+  getName(): string {
+    return "Slider 10";
+  }
+
+  sliderNext = () => {
+    const updatedItems = this.getComponentState("slider-items").map((item: SliderItem) => {
+      if (item.index === this.getComponentState("slider-items").length - 1)
+        return {
+          ...item,
+          index: 0
+        }
+      else
+        return {
+          ...item,
+          index: item.index + 1
+        }
+    })
+    this.setComponentState("slider-items", updatedItems)
+  }
+
+  sliderPrev = () => {
+    const updatedItems = this.getComponentState("slider-items").map((index: SliderItem) => {
+      if (index.index === 0)
+        return {
+          ...index,
+          index: this.getComponentState("slider-items").length - 1
+        }
+      else
+        return {
+          ...index,
+          index: index.index - 1
+        }
+    })
```

Comment:
In the same way, the code looks similar. Perhaps this code block should be a function?